code/datums/components/mood.dm

```diff
@@ -34,6 +34,13 @@
 /datum/component/mood/Destroy()
 	STOP_PROCESSING(SSmood, src)
 	unmodify_hud()
+	for(var/i in mood_events)
+		var/datum/mood_event/event = mood_events[i]
+		event.owner = null
+		if(event.timer)
+			deltimer(event.timer)
+		qdel(event)
+	mood_events.Cut()
```

Comment:
Wouldn't it be better to `deltimer` in the `mood_event`'s `Destroy`, and `QDEL_LIST_ASSOC_VAL` here?